js/PDBLoaderImproved.js

```diff
@@ -310,14 +317,20 @@ catch(e){
 			var stickGeometry = new THREE.CylinderGeometry(0.07, 0.07, 1, 3, 1, true);
 			stickGeometry.rotateX(Math.PI/2);
 			var ballGeometry = new THREE.BoxGeometry(0.3,0.3,0.3);
+			var neighborPosition = new THREE.Vector3();
+
 
 			// loop over all atoms in the molecule
-			molecule.atoms.forEach(function(atom, i)
+			for (var i = 0; i < molecule.atoms.length; i++)
 			{
 				/*
 				* Generate atom balls
 				*/
+				var atom = molecule.atoms[i];
+				var atomSeqRes = atom.seqRes;
+				var atomChainID = atom.chainID;
 				var e = atom.element.toLowerCase();
+				var atomCovalentRadius = covalentRadius[e];
```

Comment:
Alias members with variables so that the JIT compiler can optimize better inside the nested loop.